### DISCUSSION TITLE
Is LibraryImport from .Net 7 on your radar already?

### DISCUSSION THREAD

**Discussion by Marv51** [microsoft/CsWin32#647] [Q&A] [answered]
In the upcoming .NET 7 instead of DllImport a new LibraryImport attribute can be used to enable a source generator to pre-create some needed interop code.

Are you already familiar with that? Are you planning to use that for CsWin32?

I believe that chaining source generator is somewhat challenging, so this might not even be really possible?

**Comment by AArnott** [comment, marked as answer]
See discussion at https://github.com/microsoft/CsWin32/issues/593

**Reply by Marv51** [reply]
Sorry @AArnott. I did use the search, somehow I missed that.

**Reply by AArnott** [reply]
No problem. It took me a while to find it as well. The key was to search `LibraryImportAttribute` instead of `LibraryImport`.
